### PR TITLE
Migrate events and logs views to gRPC

### PR DESCRIPTION
**Pull request by vcheung-stripe**
### Summary
related to #247 

Migrate the Events and Logs views from reading stdout to reading from the gRPC stream. This reduces tech debt. This PR should be a noop.

Note that this will NOT be backwards compatible with <1.6.0 of the CLI. Users will have to upgrade their CLI to do events and logs streaming.

### Testing
Added tests for `stripeLogsView` and `stripeEventsView`, updated tests for `stripeClient`.

I verified this manually:
- For both views:
  - stream can start
  - stream can stop
  - tree items are rendered
  - tree items can be cleared
  - hovering shows timestamp
  - prompts login if you're not logged in
  - prompts upgrade if you don't have the daemon command
  - command palette commands work
- For events:
  - clicking a tree item opens the event details
  - right-clicking allows opening the event in the dashboard
- For logs:
  - clicking a tree item allows opening the log in the dashboard
  - right-clicking allows opening the log in the dashboard

### CODE REVIEW

**Comment by vcheung-stripe** [comment]
@gracegoo-stripe I hope this isn't too hard to review. Most of the logic is similar between the two views, and the tests are almost the exact same.

**Review by gracegoo-stripe — COMMENTED** [review]
LGTM! Just have one question about the reconnect case

File: src/stripeEventsView.ts [hunk 1]
```diff
@@ -70,50 +77,81 @@ export class StripeEventsViewProvider extends StreamingViewDataProvider {
...
+      case ListenResponse.State.STATE_DONE:
+        this.setViewState(ViewState.Idle);
+        break;
+      case ListenResponse.State.STATE_LOADING:
+        this.setViewState(ViewState.Loading);
+        break;
+      case ListenResponse.State.STATE_READY:
+        this.setViewState(ViewState.Streaming);
+        break;
+      case ListenResponse.State.STATE_RECONNECTING:
```
[the remark this answers is not in the document]
Comment (vcheung-stripe) [reply src/stripeEventsView.ts:124]: Oh yeah, that's probably better than not changing state. I'll change this.

**Comment by vcheung-stripe** [comment]
@gracegoo-stripe I made RECONNECTING transition to the idle state and added tests for all the state transitions.

**Review by gracegoo-stripe — APPROVED** [review]

**Pull request merged**